detox/android/detox/src/full/java/com/wix/detox/espresso/MultiViewActionWithResult.kt

```diff
@@ -0,0 +1,45 @@
+package com.wix.detox.espresso
+
+import android.view.View
+import androidx.test.espresso.Espresso
+import org.hamcrest.Matcher
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+abstract class MultiViewActionWithResult<R> : ViewActionWithResult<R> {
```

Comment:
So overall, this design is okay - but it's not good enough. There's some smell coming from the fact that the `getResult()` method is misused; Loosely speaking, one could say it violates the [Liskov Substitution Principle](https://blog.knoldus.com/what-is-liskov-substitution-principle-lsp-with-real-world-examples/#:~:text=Simply%20put%2C%20the%20Liskov%20Substitution,the%20objects%20of%20our%20superclass.).

Let's discuss where _conceptually_ that smell could be coming from.